dissect/target/plugins/os/windows/cim.py

```diff
@@ -13,13 +15,84 @@
 
     from dissect.target.target import Target
 
-ConsumerBindingRecord = TargetRecordDescriptor(
-    "filesystem/windows/cim/consumerbinding",
+# https://learn.microsoft.com/en-us/windows/win32/wmisdk/--eventconsumer
+COMMON_ELEMENTS = [
+    ("string", "creator_sid"),
+    ("string", "machine_name"),
+    ("string", "filter_query"),
+    ("string", "filter_name"),
+    ("string", "filter_query_language"),
+    ("string", "filter_creator_sid"),
+]
+
+CommandLineEventConsumerRecord = TargetRecordDescriptor(
+    "filesystem/windows/cim/consumerbinding/commandlineeventconsumer",
     [
-        ("string", "query"),
+        *COMMON_ELEMENTS,
+        ("string", "command_line_template"),
+        ("string", "executable_path"),
+        ("string", "working_directory"),
     ],
 )
 
+# https://learn.microsoft.com/en-us/windows/win32/wmisdk/activescripteventconsumer
+ActiveScriptEventConsumerRecord = TargetRecordDescriptor(
+    "filesystem/windows/cim/consumerbinding/activescripteventconsumer",
+    [
+        *COMMON_ELEMENTS,
+        ("string", "script_text"),
+        ("string", "script_file_name"),
+        ("string", "scripting_engine"),
+        ("string", "name"),
+    ],
+)
+
+
+@dataclass
+class EventFilter:
+    """Extracted information from ``__EventFilter``.
+
+    References:
+        - https://learn.microsoft.com/en-us/windows/win32/wmisdk/--eventfilter
+    """
+
+    filter_name: str = ""
+    filter_query: str = ""
+    filter_query_language: str = ""
+    filter_creator_sid: str = ""
+
+
+def get_property_value_safe(consumer: cim.Instance, prop_name: str, default_value: str | None = None) -> str | None:
+    """Extract value of a consumer properties. Fallback to ``default_value`` if property is missing."""
+    if not (prop := consumer.properties.get(prop_name)):
+        return default_value
+    try:
+        return str(prop.value)
+    except ValueError:
+        return default_value
+
+
+def get_filter_name(binding: cim.Instance) -> str:
+    """Return unquoted filter name from a ``__filtertoconsumerbinding`` class instance."""
+    filter_name = binding.properties["Filter"].value
+    # filter name is not always consistent
+    # e.g : __EventFilter.Name="Windows Update Event MOF" or
+    # \\.\root\subscription:__EventFilter.Name="Windows Update Event MOF"
+    if "=" in filter_name:
+        # Required to manage filters name with escaped "
+        _, _, filter_name = filter_name.partition("=")
+        filter_name = filter_name.strip('"').replace('\\"', '"')
+    return filter_name
+
+
+def get_creator_sid(class_instance: cim.Instance) -> str | None:
+    """Extract and parse ``CreatorSID`` member, if available."""
+    if creator_sid := class_instance.properties.get("CreatorSID"):
+        creator_sid_value = getattr(creator_sid, "value", None)
+        if creator_sid_value:
+            return read_sid(bytes(creator_sid_value))
```

Comment:
```suggestion
    if (creator_sid := class_instance.properties.get("CreatorSID")) and (creator_sid_value := getattr(creator_sid, "value", None)):
        return read_sid(bytes(creator_sid_value))
```
I'm not sure if the walrus operator is this flexible but I think it is.